render/canvas/src/lib.rs

```diff
@@ -451,7 +456,15 @@ impl RenderBackend for WebCanvasRenderBackend {
         None
     }
 
-    fn submit_frame(&mut self, clear: Color, commands: CommandList) {
+    fn submit_frame(
+        &mut self,
+        clear: Color,
+        commands: CommandList,
+        cache_entries: Vec<BitmapCacheEntry>,
+    ) {
+        if !cache_entries.is_empty() {
+            panic!("Bitmap caching is unavailable on the canvas backend");
```

Comment:
Should log an error instead of panicking?

Reply:
No, we shouldn't be able to get here and if we did a developer messed up (and will get this during development). You'll get graphical corruption if you proceed anyway, because it'll be using a texture of uninitialized values.